### PR TITLE
image upload

### PR DESCRIPTION
**Pull request by NicolasBonduel**
This PR adds the possibility of uploading images through the SDK.

### CODE REVIEW

File: src/collection/resources/_resources.js [hunk 1]
```diff
@@ -34,13 +34,14 @@ class Components {
       apiPath = this.apiPath,
       action = '',
       ResConstructor = this.ResConstructor,
+      isJson = true,
       json,
```
Comment (NicolasBonduel) [inline src/collection/resources/_resources.js]: I'm not especially happy with this naming, as we now have an `isJson` and a `json` parameter in this function.
The `json` parameter allows for choosing if we would like an object or a json response from the promise/callback.
The `isJson` parameter allows for choosing if we should `JSON.stringify` the body before sending, and what `Content-Type` we should apply to the request.

Comment (lazreg87) [reply src/collection/resources/_resources.js]: hm, maybe it's better to have a content_type parameter, that defaults to application/json?

File: src/collection/resources/images.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const FormData = require('form-data');
+const Components = require('./_resources');
+const Image = require('../objects/image');
+
+class PaymentMethods extends Components {
```
Comment (lazreg87) [inline src/collection/resources/images.js]: PaymentMethods?

File: src/utils/request.js [hunk 3]
```diff
@@ -73,23 +73,24 @@ class Request {
       return this.getHeaders().then((queryHeaders) => {
         request[type](`${endpoint}${path}${Request.serialize(urlParams)}`)
           .set(Object.assign({}, queryHeaders, headers))
-          .send(JSON.stringify(body))
+          .set(isJson ? { 'Content-Type': 'application/json' } : {}) // when undefined, SuperAgent guesses it.
```
Comment (lazreg87) [inline src/utils/request.js]: look at comment above, would make this more clear as well

File: src/collection/resources/_resources.js (hunk 1 above)
Comment (NicolasBonduel) [reply src/collection/resources/_resources.js]: Yeah, that's what I thought, but the content type we need to send looks like this:
`multipart/form-data; boundary=---------------------------1907295027419120020403813980`
Much easier to let superagent deal with it, no?

Comment (tigo0) [reply src/collection/resources/_resources.js]: I think the browser is adding the `boundary` stuff automatically if it recognizes the mutipart.

**Pull request merged**